bootstrap/041_quotas.sql

```diff
@@ -30,22 +30,116 @@ costed_queries as (
         role_name
     from todays_queries
 ),
--- Is there a way to collapse user_usage and role_usage together? They're essentially the same query.
-user_usage as (
-    select user_name as name, sum(credits_used) as usage_map
+usage as (
+    select user_name as name, sum(credits_used) as credits_used, 'user' as persona
     from costed_queries
     group by user_name
-),
-role_usage as (
-    select role_name as name, sum(credits_used) as usage_map
+    union all
+    select role_name as name, sum(credits_used) as credits_used, 'role' as persona
     from costed_queries
     group by role_name
-),
-usage_aggr as (
-    select 'users' as kind, object_agg(name, usage_map) as credits_used from user_usage union all select 'roles', object_agg(name, usage_map) from role_usage
 )
--- Aggregate into a final map {'users': {..}, 'roles': {..}}
-select object_agg(kind, credits_used) as daily_quota from usage_aggr;
+select name, persona, credits_used from usage;
 $$;
     return s;
 END;
+
+create table if not exists internal.aggregated_hourly_quota(hour_of_day integer, name string, persona string, credits_used float, last_updated timestamp);
```

Comment:
I am not sure I like this. why not just use the table function? then we don't have to mess around w/ edge cases. At the restriction that 10k rows in an hour max. Thats a good trade off to avoid the below code.